Codify

darcymason opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/pydicom/pydicom/blob/master/CONTRIBUTING.md#contributing-pull-requests
-->
#### Reference Issue
Issue #498


#### What does this implement/fix? Explain your changes.
* fixes long/int python3 problem when run as script
* adds new OD and OL dicom VRs as byte types
* fixes codify script bug when data element value has no len()
* updates so writes file_meta code using write_like_original=False

#### Any other comments?
<!--
-->
Will add documentation for `codify` in separate PR
<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.

Thanks for contributing!
-->

pep8speaks commented:
Hello @darcymason! Thanks for submitting the PR.



 - In the file [`pydicom/util/codify.py`](https://github.com/pydicom/pydicom/blob/4db41a9a083ae36d47dc85bb5ab777b4ea731709/pydicom/util/codify.py), following are the PEP8 issues :

> [Line 344:80](https://github.com/pydicom/pydicom/blob/4db41a9a083ae36d47dc85bb5ab777b4ea731709/pydicom/util/codify.py#L344): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 79 characters)

mrbean-bremen commented on pydicom/util/codify.py:
```diff
@@ -33,9 +34,15 @@
+else:
+    int_type = int
+
 
```
maybe move this to `compat.py` where `number_types` if defined?

darcymason replied:
Ah, you caught me - I had the same thought, but then was too lazy to follow through.  I'll do it.

mrbean-bremen commented on pydicom/util/codify.py:
```diff
@@ -333,8 +341,10 @@ def code_file(filename, exclude_size=None, include_private=False):
     if args.save_as:
-        save_line = "\nds.save_as('{filename}')".format(filename=args.save_as)
+        msg = "\nds.save_as('{filename}', write_like_original=False)"
+        save_line = msg.format(filename=args.save_as)
```
better make `'{filename}'` unicode (e.g. `u'{filename}'`), in case the file name is not ASCII

darcymason commented:
So, added and moved around a bit.  Now, the ds.save_as code line is always created, but with an auto-generated name if the user doesn't supply one.  Moved the `if __name__ == "__main__"` script lines to a main() function, and added a unit test for that function.
